src/globalstate.h

```diff
@@ -18,9 +18,30 @@
 #include <src/gasman.h>
 
 #define MAXPRINTDEPTH 1024L
+#define TLS_NUM_EXTRA 256
 
-typedef struct GlobalState
+typedef struct GAPState
 {
+#if defined(HPCGAP)
+  int threadID;
+  void *threadLock;
+  void *threadSignal;
+  void *acquiredMonitor;
+  unsigned multiplexRandomSeed;
```

Comment:
I know you just copied this -- but I think using a "naked" `unsigned` like that is a bad idea, I'd change that to `unsigned int` -- or possibly a suitable type from `inttypes.h`